pkg/apis/networking/validation/validation_test.go

```diff
@@ -33,1022 +33,493 @@ import (
+		networkPolicy.Spec.Egress[0].Ports = []networking.NetworkPolicyPort{
+			{
+				Protocol: &protocolSCTP,
+				Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 33000},
+				EndPort:  &endPort,
 			},
-		},
-		"multiple invalid port ranges defined": {
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Egress: []networking.NetworkPolicyEgressRule{
-					{
-						To: []networking.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-							},
-						},
-						Ports: []networking.NetworkPolicyPort{
-							{
-								Protocol: &protocolUDP,
-								Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 35000},
-								EndPort:  &endPort,
-							},
-							{
-								Protocol: &protocolTCP,
-								EndPort:  &endPort,
-							},
-							{
-								Protocol: &protocolTCP,
-								Port:     &intstr.IntOrString{Type: intstr.String, StrVal: "https"},
-								EndPort:  &endPort,
-							},
-						},
-					},
-				},
+		}
+	}
+
+	setMultipleInvalidPortRanges := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Egress[0].Ports = []networking.NetworkPolicyPort{
+			{
+				Protocol: &protocolUDP,
+				Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 35000},
+				EndPort:  &endPort,
+			},
+			{
+				Protocol: &protocolTCP,
+				EndPort:  &endPort,
+			},
+			{
+				Protocol: &protocolTCP,
+				Port:     &intstr.IntOrString{Type: intstr.String, StrVal: "https"},
+				EndPort:  &endPort,
 			},
-		},
-		"invalid endport range defined": {
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Egress: []networking.NetworkPolicyEgressRule{
-					{
-						To: []networking.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-							},
-						},
-						Ports: []networking.NetworkPolicyPort{
-							{
-								Protocol: nil,
-								Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 30000},
-								EndPort:  utilpointer.Int32Ptr(65537),
-							},
-						},
-					},
-				},
+		}
+	}
+
+	setInvalidEndPortRanges := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Egress[0].Ports = []networking.NetworkPolicyPort{
+			{
+				Protocol: nil,
+				Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 30000},
+				EndPort:  utilpointer.Int32Ptr(65537),
 			},
-		},
+		}
+	}
+
+	errorCases := map[string]*networking.NetworkPolicy{
+		"namespaceSelector and ipBlock":                     makeNetworkPolicyCustom(setIngressEmptyFirstElement, setIngressFromEmptyFirstElement, setIngressFromNamespaceSelector, setIngressFromIPBlock),
```

Comment:
This certainly shortens the test, and I think it's MOSTLY clearer.  It's also OK to mix the modes when they are very localized.

e.g.

```
"invalid ingress.ports.protocol":  makeNetworkPolicyCustom(func(networkPolicy *networking.NetworkPolicy) {
	networkPolicy.Spec.Ingress = makeEmptyIngress()
	networkPolicy.Spec.Ingress[0].Ports = []networking.NetworkPolicyPort{{
		Protocol: &protocolICMP,
		Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 80},
	}}
})
```

or even

```
e.g.

```
"invalid ingress.ports.protocol":  makeNetworkPolicyCustom(func(networkPolicy *networking.NetworkPolicy) {
	networkPolicy.Spec.Ingress = makeValidIngress()
	networkPolicy.Spec.Ingress[0].Ports[0].Protocol = &protocolICMP
```

The goal being that the reader of the testcase can see quickly "what is special
about this case".